services/relayer/internal/msgdispatcher/eth/mq/handler_v1.go

```diff
@@ -53,50 +56,100 @@ func handleV1(req messages.Message) {
+			return
+		}
+		signature, err := hex.DecodeString(msg.Proofs[0].Proof)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
 
-	link, err := instance.Transactor.GetChainAP(big.NewInt(int64(destID)))
-	if err != nil {
-		logging.Error(err.Error())
-		return
-	}
-	auth, err := instance.Transactor.GetAuth(big.NewInt(int64(destID)))
-	if err != nil {
-		logging.Error(err.Error())
-		return
-	}
-	esAddr, err := instance.Verifier.GetVerifierAddr(big.NewInt(int64(destID)), destAddr)
-	if err != nil {
-		logging.Error(err.Error())
-		return
-	}
-	logging.Info("Obtain event store address: %v", esAddr.String())
+		link, err := instance.Transactor.GetChainAP(big.NewInt(int64(destID)))
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		auth, err := instance.Transactor.GetAuth(big.NewInt(int64(destID)))
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		esAddr, err := instance.Verifier.GetVerifierAddr(big.NewInt(int64(destID)), destAddr)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		logging.Info("Obtain event store address: %v", esAddr.String())
 
-	logging.Info("Adding message %v to queue for process...", msg.ID)
-	instance.Dispatcher.AddToQueue(link, auth, msg.ID, big.NewInt(int64(destID)), esAddr, big.NewInt(int64(srcID)), srcAddr, raw.Data, signature)
-	logging.Info("Message %v is added to queue.", msg.ID)
+		logging.Info("Adding message %v to queue for process...", msg.ID)
+		instance.Dispatcher.AddToQueue(link, auth, msg.ID, big.NewInt(int64(destID)), esAddr, big.NewInt(int64(srcID)), srcAddr, raw.Data, signature)
+		logging.Info("Message %v is added to queue.", msg.ID)
+	} else {
+		logging.Info("Received message for bridging from contract %v on chain %v to message store on %v", srcAddr.String(), srcID, instance.MessageStoreAddr)
+
+		data, err := hex.DecodeString(msg.Payload)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		raw := types.Log{}
+		err = json.Unmarshal(data[32:], &raw)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		proofs, err := json.Marshal(msg.Proofs)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		// Put message
+		client := &http.Client{}
+		req, err := http.NewRequest("PUT", fmt.Sprintf("http://%v/messages/", instance.MessageStoreAddr), bytes.NewReader(msg.ToBytes()))
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		resp, err := client.Do(req)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		if !(resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusCreated) {
+			logging.Error("Error creating new message: %v", resp.Status)
+			return
+		}
+
+		req, err = http.NewRequest("PUT", fmt.Sprintf("http://%v/messages/%v/proofs/", instance.MessageStoreAddr, msg.ID), bytes.NewReader(proofs))
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
+		resp, err = client.Do(req)
+		if err != nil {
+			logging.Error(err.Error())
+			return
+		}
```

Comment:
I think this request would be redundant because the earlier `PUT` in line 126, would already update the proofs? 
From the [API spec for PUT /messages](https://github.com/ConsenSys/gpact/blob/main/services/message-store/api/message-store-service-api.yml#L73-L78): 
```
 The endpoint is used to submit (and create) a message if it does not
 already exist in the store. If it does, the existing message is updated
 to include the additional proof data of the submitting Relayer...
```